zipline/data/data_portal.py

```diff
@@ -349,14 +345,13 @@ def get_history_window(self, sids, end_dt, bar_count, frequency, field,
             data = []
 
             day = end_dt.date()
-
-            # for daily history, we need to get bar_count - 1 complete days,
-            # and then a partial day constructed from the minute data of this
-            # day.
             day_idx = tradingcalendar.trading_days.searchsorted(day)
             days_for_window = tradingcalendar.trading_days[
                 (day_idx - bar_count + 1):(day_idx + 1)]
 
+            ends_at_midnight = end_dt.hour == 0 and end_dt.minute == 0 \
```

Comment:
I think this shows a need to separate out two methods `get_daily_history` and `get_minute_history`, which can be dispatched/routed on the `frequency` flag. Should be more debuggable etc. than handling the special case throughout the '1m' case.